ddtrace/_monkey.py

```diff
@@ -118,15 +121,19 @@ def _on_import_factory(module, raise_errors=True):
     def on_import(hook):
         # Import and patch module
         path = "ddtrace.contrib.%s" % module
+        integration = create_integration(module)
         try:
             imported_module = importlib.import_module(path)
-        except ImportError:
+        except ImportError as ie:
+            on_exception(integration, ie.msg, "no")
```

Comment:
I wonder if it is better to have `create_integration` also take in any exception?

having a random method `on_exception` to attach metadata to a dict is a little weird to read, but if we had instead:

```
queue_integration(create_integration(module, exception, ...))
```